display_list/display_list_color_filter.h

```diff
@@ -57,6 +57,11 @@ class DlColorFilter
   // pixels non-transparent and therefore expand the bounds.
   virtual bool modifies_transparent_black() const = 0;
 
+  // Return a boolean indicating whether the color filtering operation can
+  // be applied either before or after modulating the pixels with an opacity
+  // value without changing the operation.
+  virtual bool can_commute_with_alpha() const { return false; }
```

Comment:
Here, if mostly CF can commute with alpha, but if the CF is a MtrixCF and has specific alpha in the matrix it should not commute with alpha?